analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AssignmentExpressionSyntaxExtensionsTests.cs

```diff
@@ -27,6 +27,356 @@ namespace SonarAnalyzer.UnitTest.Extensions
     [TestClass]
     public class AssignmentExpressionSyntaxExtensionsTests
     {
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_MapAssignmentArguments_TupleElementsAreExtracted()
+        {
+            var code = "(var x, var y) = (1, 2);";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
+                    new
+                    {
+                        Left = new { Designation = new { Identifier = new { Text = "x" } } },
+                        Right = new { Token = new { Text = "1" } },
+                    },
+                    new
+                    {
+                        Left = new { Designation = new { Identifier = new { Text = "y" } } },
+                        Right = new { Token = new { Text = "2" } },
+                    },
+                });
+        }
+
+        [TestMethod]
+        public void AssignmentExpressionSyntaxExtensions_MapAssignmentArguments_SimpleAssignmentReturnsSingleElementArray()
+        {
+            var code = "int x; x = 1;";
+            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
+            mapping.Should().BeEquivalentTo(new[]
+                {
```

Comment:
This part repeats a lot, let's extract it to a method with `string code, params AssignmentMapping[] expected)` signature.
```suggestion
            var mapping = ParseAssignmentExpression(code).MapAssignmentArguments();
            mapping.Should().BeEquivalentTo(new[]
                {
```
The test would look like
public void Something() =>
    AssertMapAssignmentArguments(
        "code....;",
        new
        {
            ... same as before
        });